inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/hive/HiveSink.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    /**
+     * configure
+     * 
+     * @param context
+     */
+    @Override
+    public void configure(Context context) {
+        LOG.info("start to configure:{}, context:{}.", this.getClass().getSimpleName(), context.toString());
+        this.parentContext = context;
+    }
+
+    /**
+     * process
+     * 
+     * @return                        Status
+     * @throws EventDeliveryException
+     */
+    @Override
+    public Status process() throws EventDeliveryException {
+        Channel channel = getChannel();
+        Transaction tx = channel.getTransaction();
+        tx.begin();
+        try {
+            Event event = channel.take();
+            if (event == null) {
+                tx.commit();
+                return Status.BACKOFF;
+            }
+            if (!(event instanceof ProfileEvent)) {
+                tx.commit();
+                this.context.addSendFailMetric();
+                return Status.READY;
+            }
+            //
+            ProfileEvent profileEvent = (ProfileEvent) event;
+            this.dispatchManager.addEvent(profileEvent);
+            tx.commit();
+            return Status.READY;
+        } catch (Throwable t) {
+            LOG.error("Process event failed!" + this.getName(), t);
+            try {
+                tx.rollback();
+            } catch (Throwable e) {
+                LOG.error("Channel take transaction rollback exception:" + getName(), e);
+            }
+            return Status.BACKOFF;
+        } finally {
+            tx.close();
+        }
+    }
+
+    /**
+     * writeHdfsFile
+     */
+    private void writeHdfsFile() {
+        // write file
+        long currentTime = System.currentTimeMillis();
+        DispatchProfile dispatchProfile = this.dispatchQueue.poll();
+        while (dispatchProfile != null) {
+            String uid = dispatchProfile.getUid();
+            HdfsIdConfig idConfig = context.getIdConfigMap().get(uid);
+            if (idConfig == null) {
+                // monitor 007
+                LOG.error("can not find uid:{},idConfigMap:{}", uid, JSON.toJSONString(context.getIdConfigMap()));
+                this.context.addSendResultMetric(dispatchProfile, uid, false, 0);
+                dispatchProfile = this.dispatchQueue.poll();
+                continue;
+            }
+            String strIdRootPath = idConfig.parsePartitionPath(dispatchProfile.getDispatchTime());
+            HdfsIdFile idFile = this.hdfsIdFileMap.get(strIdRootPath);
+            if (idFile == null) {
+                try {
+                    idFile = new HdfsIdFile(context, idConfig, strIdRootPath);
+                } catch (Exception e) {
+                    // monitor 007
+                    LOG.error("can not connect to hdfsPath:{},write file:{}", context.getHdfsPath(), strIdRootPath);
+                    this.context.addSendResultMetric(dispatchProfile, uid, false, 0);
+                    dispatchProfile = this.dispatchQueue.poll();
```

Comment:
Here is another message, will it cause the loss of the previous message?

Reply:
The config of this id is bad, so missing the data of this id.